src/RandMsg.chpl

```diff
@@ -572,70 +573,117 @@ module RandMsg
         // uses the algorithm from knuth found here:
         // https://en.wikipedia.org/wiki/Poisson_distribution#Random_variate_generation
         // generates values drawn from poisson distribution using a stream of uniformly distributed random numbers
-
-        const nTasksPerLoc = here.maxTaskPar; // tasks per locale based on locale0
-        const Tasks = {0..#nTasksPerLoc}; // these need to be const for comms/performance reasons
-
-        const generatorSeed = (rng.next() * 2**62):int;
         var poissonArr = makeDistArray(size, int);
 
-        // I hate the code duplication here but it's not immediately obvious to me how to avoid it
-        if isSingleLam {
-            const lam = lamStr:real;
-            // using nested coforalls over locales and tasks so we know how to generate taskSeed
-            coforall loc in Locales do on loc {
-                const generatorIdxOffset = here.id * nTasksPerLoc,
-                    locSubDom = poissonArr.localSubdomain(),  // the chunk that this locale needs to handle
-                    indicesPerTask = locSubDom.size / nTasksPerLoc;  // the number of elements each task needs to handle
-
-                coforall tid in Tasks {
-                    const taskSeed = generatorSeed + generatorIdxOffset + tid,  // initial seed offset by other locales threads plus current thread id
-                        startIdx = tid * indicesPerTask,
-                        stopIdx = if tid == nTasksPerLoc - 1 then locSubDom.size else (tid + 1) * indicesPerTask;  // the last task picks up the remainder of indices
-                    var rs = new randomStream(real, taskSeed);
-                    for i in startIdx..<stopIdx {
-                        var L = exp(-lam);
-                        var k = 0;
-                        var p = 1.0;
-
-                        do {
-                            k += 1;
-                            p = p * rs.next(0, 1);
-                        } while p > L;
-                        poissonArr[locSubDom.low + i] = k - 1;
+        if hasSeed {
+            // use a fixed number of elements per stream instead of relying on number of locales or numTasksPerLoc because these
+            // can vary from run to run / machine to mahchine. And it's important for the same seed to give the same results
+            const generatorSeed = (rng.next() * 2**62):int,
+                elemsPerStream = 4096;
```

Comment:
yah I def think this is a good idea and would like us to have some sort of dynamicyness in this PR. My thought along the same lines as this is to look use the order of magnitude of the size to scale the order of magnitude of elemsPerStream

I think for this first pass, instead of table I'll try to do some sorta direct calculation and then we can fine tune it in the future. Maybe something like

```chapel
min_per_stream = 256,
elemsPerStream = max(min_per_stream, 2**(2 * log10(size));
```
plus or mins a cast to int and/or ceil/floor

Reply:
yeah that sounds like a good way to do it. Probably better than a lookup table imo